packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -79,7 +79,7 @@ function isTemplateOnly(matches: string[], filePath: string) {
 
   let correspondingFileGlob = path.join(
     path.dirname(filePath),
-    path.basename(filePath).replace(/hbs$/, '*')
+    `{${path.basename(filePath).replace(/hbs$/, '*')},route.*,controller.*}`
```

Comment:
for example:
```diff
    addon.appReexports([
      'components/**/*.js',
      'helpers/**/*.js',
      'modifiers/**/*.js',
      'services/**/*.js',
+     'routes/**/*.js',
+     'controller/**/*.js',
    ]),
```

Reply:
I don’t think that would help here, since template files will still be considered template only, even if the route and/or controller files are present. 

The issue is that the route/controller template files are being compiled as template only files, which is incorrect. This change prevents them from being compiled as such (just like co-located templates, which this function tests for)

Reply:
can we update the test to use this style of rollup config, and see if we can back out the routes/controllers glob?

I think it's maybe just https://github.com/embroider-build/embroider/pull/1640/files#diff-b40fbc924d5845e9b6c9ea00865362929e875d19daf98cc8752a473696333fcfR25 this change you need?

Reply:
There's also this approach: https://github.com/discourse/ember-route-template

Reply:
Yeah, `ember-route-template` is the best path forward, but unfortunately it doesn't help us when we want to refactor our current monolithic ember app into engines/addons unless we want to spend more time refactoring templates as well (some of which are quite large!)

Anyway, removing this particular change doesn't seem to work, since there end up being two `template` files after compilation. We're using a pods structure, so here's the rollup config and the output during the build phase:

```
addon.publicEntrypoints([
  'pods/**/*.js',
  'pods/**/template.hbs',
  'components/**/*.js',
  'router.js',
  'index.js',
  'template-registry.js',
]),

n.appReexports([
  'components/**/*.js',
  'pods/**/*.js',
  'pods/**/template.hbs',
]),
```

```
./packages/workspace-settings/addon/dist
├── _app_
│   ├── components
│   │   ├── a.js
│   │   └── b.js
│   └── pods
│       └── workspace-settings
│           ├── edit
│           │   ├── route.js
│           │   ├── template.hbs
│           │   └── template.js
│           ├── index
│           │   ├── route.js
│           │   ├── template.hbs
│           │   └── template.js
│           ├── route.js
│           ├── template.hbs
│           └── template.js
├── a-3761c7ca.js
├── a-3761c7ca.js.map
├── a-b89cc467.js
├── a-b89cc467.js.map
├── b-1571e635.js
├── b-1571e635.js.map
├── b-8aef0527.js
├── b-8aef0527.js.map
├── components
│   ├── a.js
│   ├── a.js.map
│   ├── b.js
│   └── b.js.map
├── index.js
├── index.js.map
├── pods
│   └── workspace-settings
│       ├── edit
│       │   ├── route.js
│       │   ├── route.js.map
│       │   ├── template.hbs
│       │   ├── template.js
│       │   └── template.js.map
│       ├── index
│       │   ├── route.js
│       │   ├── route.js.map
│       │   ├── template.hbs
│       │   ├── template.js
│       │   └── template.js.map
│       ├── route.js
│       ├── route.js.map
│       ├── template.hbs
│       ├── template.js
│       └── template.js.map
├── router.js
├── router.js.map
├── template-registry.js
└── template-registry.js.ma
```

As you can see, each route template gets exported twice in `_app_`: `template.hbs` and `template.js`. When running the test app, however, we get our first problems due to the existance of both template files:

```
npm run start -- --port 4400

> test-app@0.0.0 start
> ember serve --port 4400

Build Error (TemplateCompiler)

EEXIST: file already exists, symlink '/Users/mike/projects/ui-monorepo/packages/workspace-settings/addon/dist/_app_/pods/workspace-settings/edit/template.js' -> '/var/folders/_n/fmcdhjh14gb1jtxshg6tcqgr0000gn/T/broccoli-72988GIV8sQIDekxA/out-276-template_compiler/test-app/pods/workspace-settings/edit/template.js'
```

Of course if I change the build config to this, it works.

```
addon.appReexports([
  'components/**/*.js',
  'pods/**/route.js',
  'pods/**/controller.js',
  'pods/**/template.hbs',
]),
```

In the end, I still feel like a function that's testing if a handlebars file is a template-only component, it should take into account the possibility of it being a route template in addition to being a co-located template (which this change extends from).